### PR TITLE
GEODE-10216: Add test for existence of VM stats

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
When we started running with JDK 11, some users noticed that a few stats
had gone missing. There was a one line fix in test.gradle, but the
change was not tested. This adds a test for the existence of the VM
stats so that none of the stats go missing in the future.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java [hunk 1]
```diff
@@ -490,6 +490,53 @@ public void testPubAndSubCustomStats() throws Exception {
...
+        vmStats.get("threadStarts");
+        vmStats.get("cpus");
+        vmStats.get("freeMemory");
+        vmStats.get("totalMemory");
+        vmStats.get("maxMemory");
+        vmStats.get("processCpuTime");
+        vmStats.get("fdsOpen");
+        vmStats.get("fdLimit");
+      } catch (IllegalArgumentException e) {
+        fail("Stats do not exist: " + e.getMessage());
```
Comment (pivotal-jbarrett) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java:535]: I would love to see us shy away from the this style of testing where we catch exceptions and then `fail`. I prefer that we either let the exception be thrown, and that in itself fails the tests or AssertJ's `assertThatNoException().isThrownBy(() -> {something()})`.

**Review by pivotal-jbarrett — APPROVED** [review]

File: geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java (hunk 1 above)
Comment (nonbinaryprogrammer) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java]: thank you! I was trying to use `assertThatThrownBy(() -> something).doesNotThrow()` and it was failing when the stat exists/no exception is thrown

File: geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java [hunk 2]
```diff
@@ -490,6 +491,49 @@ public void testPubAndSubCustomStats() throws Exception {
...
+    vm.invoke(() -> {
+      Properties props = new Properties();
+      props.setProperty(STATISTIC_SAMPLING_ENABLED, "true");
+      props.setProperty(STATISTIC_SAMPLE_RATE, "1000");
+
+      InternalDistributedSystem system = getSystem(props);
+
+      // assert that sampler is working as expected
+      GemFireStatSampler sampler = system.getStatSampler();
+      assertTrue(sampler.isSamplingEnabled());
```
Comment (pivotal-jbarrett) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java:508]: We should make sure the new test is at least written in AssertJ. If the effort to convert the other tests is small please consider that.

**Review by pivotal-jbarrett — CHANGES_REQUESTED** [review]

File: geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java (hunk 2 above)
Comment (nonbinaryprogrammer) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java:508]: ah, defeated by copy-paste. gonna fix it all

File: geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java [hunk 3]
```diff
@@ -544,10 +584,12 @@ public static void main(final String[] args) throws Exception {
...
 
-      assertTrue(statType1 + "#" + statName1 + "=" + value1 + " does not equal " + statType2 + "#"
-          + statName2 + "=" + value2, value1 == value2);
+      assertThat(value1)
+          .as(statType1 + "#" + statName1 + "=" + value1 + " does not equal " + statType2 + "#"
+              + statName2 + "=" + value2)
+          .isEqualTo(value2);
     } else {
-      assertEquals("Minimum two args are required: statType statName", 2, args.length);
+      assertThat(args.length).as("Minimum two args are required: statType statName").isEqualTo(2);
```
Comment (pivotal-jbarrett) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java:592]: Isn't there an `assertThat(args)....hasSize(2)`

File: geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java [hunk 4]
```diff
@@ -749,24 +791,20 @@ int readIntStat(final String typeName, final String statName) throws IOException
...
         List<File> archives = findFilesWithSuffix(dir, regex, ".gfs");
         return readIntStatFromArchives(archives, typeName, statName);
+      }
 
-        // one archive file
-      } else if (dir.exists() && dir.isFile()) {
-        List<File> archives = new ArrayList<>();
-        archives.add(dir);
-        return readIntStatFromArchives(archives, typeName, statName);
+      assertThat(dir.exists() && dir.isFile()).as("archive file should exist").isTrue();
```
Comment (pivotal-jbarrett) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java:796]: There is a `assertThat(dir).isFile()` that satisfies both exists and is a file.

**Review by pivotal-jbarrett — APPROVED** [review]

**Pull request merged**